crates/circuit/src/circuit_instruction.rs

```diff
@@ -878,20 +878,37 @@ pub fn convert_py_to_operation_type(
         Ok(stdgate) => stdgate.extract().ok().unwrap_or_default(),
         Err(_) => None,
     };
-    // If the input instruction is a standard gate and a singleton instance
+    // If the input instruction is a standard gate and a singleton instance,
     // we should check for mutable state. A mutable instance should be treated
     // as a custom gate not a standard gate because it has custom properties.
-    //
-    // In the futuer we can revisit this when we've dropped `duration`, `unit`,
+    // Controlled gates with non-default control states are also considered
+    // custom gates even if a standard representation exists for the default
+    // control state.
+
+    // In the future we can revisit this when we've dropped `duration`, `unit`,
     // and `condition` from the api as we should own the label in the
     // `CircuitInstruction`. The other piece here is for controlled gates there
     // is the control state, so for `SingletonControlledGates` we'll still need
     // this check.
     if standard.is_some() {
         let mutable: bool = py_op.getattr(py, intern!(py, "mutable"))?.extract(py)?;
-        if mutable
+        // The default ctrl_states are 1, "1" and None. These are the only cases where controlled
+        // gates can be standard.
+        let is_default_ctrl_state: bool = match py_op.getattr(py, intern!(py, "ctrl_state")) {
+            Ok(c_state) => match c_state.extract::<Option<i32>>(py) {
+                Ok(c_state_int) => match c_state_int {
+                    Some(c_int) => c_int == 1,
+                    None => true,
+                },
+                Err(_) => false,
+            },
+            Err(_) => false,
+        };
```

Comment:
Ok, so the status of things after [da3e598](https://github.com/Qiskit/qiskit/pull/12659/commits/da3e598b91c216cbddad3413990a90b3151196a9) (and [b70799a](https://github.com/Qiskit/qiskit/pull/12659/commits/b70799a4e35a722e06cc7bde18816a89acb8a725) which I needed to fix the logic that I screwed up in [05fe760](https://github.com/Qiskit/qiskit/pull/12659/commits/05fe7608072dcb146189a0ca59c13864cafb272a)) is that we check for the following conditions to skip the rust definition:

a. mutable and (singleton or controlled singleton)
b. controlled and non-default-ctrl-state (I understand there's no need to add the mutable condition here because it will always be mutable?)

Reply:
Yeah if you do things in that order it should work I think.